app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery',
+    'underscore',
+    'uiComponent',
+    'uiRegistry',
+    'Magento_Ui/js/modal/modal',
+    'Magento_Checkout/js/model/quote',
+    'Magento_Customer/js/model/customer',
+    'Magento_Checkout/js/model/step-navigator',
+    'Magento_Checkout/js/model/address-converter',
+    'Magento_Checkout/js/action/set-shipping-information',
+    'Magento_InventoryInStorePickupFrontend/js/model/pickup-locations-service',
+], function(
+    $,
+    _,
+    Component,
+    registry,
+    modal,
+    quote,
+    customer,
+    stepNavigator,
+    addressConverter,
+    setShippingInformationAction,
+    pickupLocationsService
+) {
+    'use strict';
+
+    return Component.extend({
+        defaults: {
+            template: 'Magento_InventoryInStorePickupFrontend/store-selector',
+            selectedLocationTemplate:
+                'Magento_InventoryInStorePickupFrontend/store-selector/selected-location',
+            storeSelectorPopupTemplate:
+                'Magento_InventoryInStorePickupFrontend/store-selector/popup',
+            storeSelectorPopupItemTemplate:
+                'Magento_InventoryInStorePickupFrontend/store-selector/popup-item',
+            loginFormSelector:
+                '#store-selector form[data-role=email-with-possible-login]',
+            defaultCountryId: window.checkoutConfig.defaultCountryId,
+            imports: {
+                nearbySearchRadius: '${ $.parentName }:nearbySearchRadius',
+                nearbySearchLimit: '${ $.parentName }:nearbySearchLimit',
+            },
+        },
+        selectedLocation: pickupLocationsService.selectedLocation,
+        quoteIsVirtual: quote.isVirtual(),
+        searchQuery: '',
+        nearbyLocations: null,
+        isLoading: pickupLocationsService.isLoading,
+        popup: null,
+
+        initialize: function() {
+            this._super();
+
+            var updateNearbyLocations = _.debounce(function(searchQuery) {
+                var postcode, city;
+                city = searchQuery.replace(/(\d+[\-]?\d+)/, function(match) {
+                    postcode = match;
+
+                    return '';
+                });
+
+                this.updateNearbyLocations(
+                    addressConverter.formAddressDataToQuoteAddress({
+                        city: city,
+                        postcode: postcode,
+                        country_id: quote.shippingAddress().countryId,
+                    })
+                );
+            }, 300).bind(this);
```

Comment:
Move debounce time to compoenent configuration.

Reply:
Done.